src/core/server/index.ts

```diff
@@ -311,6 +311,7 @@ export {
   exportSavedObjectsToStream,
   importSavedObjectsFromStream,
   resolveSavedObjectsImportErrors,
+  generateSavedObjectId,
```

Comment:
We should reduce non-contract APIs exposed from `core` to a minimum (the import/export static function are remains from legacy that will be moved to the SO service mid term). Any reason this `generateSavedObjectId ` is exposed as a static function instead of being provided by the `savedObjects` service contract?

Reply:
It's a pure function that doesn't require any shared state so didn't make sense to me to add it to the `SavedObjectsSerializer`. Maybe we could add it as a static method to `SavedObjectsServiceSetup` if that addresses your concern?

Reply:
cc @rudolf wdyt? is exposing `generateSavedObjectId` statically from the index acceptable to you, or do you think this should be exposed from the SO service contract?

Reply:
Another option could be to put this into `kbn-utils` package.

Reply:
> Another option could be to put this into kbn-utils package.

I don't think we should. `Separate by domain` is one of our [main principles](https://github.com/elastic/kibana/blob/2f1628784ab42372b93b8cab320d26c8baa9a9d0/src/core/PRINCIPLES.md#explicit-business-domains-separation), so it should belong to `saved objects`.
++ to add it as a `setup/start` contract property. `id creation` operation doesn't sound like `serializer` responsibility.
btw why other code in SO service still calls `uuid` to generate `id`? https://github.com/elastic/kibana/blob/d4b2a5145a9aa813146e2c587f7be2929f09fb7e/src/core/server/saved_objects/service/lib/repository.ts#L350

Reply:
> id creation operation doesn't sound like serializer responsibility.

I agree. You're stepping on some tech debt by trying to expose a standardised generate id function, but I think now's a good time to refactor this and move id generation out of the serializer.

We could change 
```
public generateRawId(namespace: string | undefined, type: string, id?: string)
 // -->
public generateRawId(namespace: string | undefined, type: string, id: string) // id is no longer optional
```

This will probably create some type failures, we would have to change `SavedObjectSanitizedDoc` to make `SavedObjectSanitizedDoc['id']` a required property.

Reply:
Thanks for suggestions everyone! ❤️ 

I have played around with putting the helper functions into the `SavedObjectService` contract but it would be a much bigger refactor than I feel comfortable changing for the purpose of this PR and add complexity, both architecturally and in terms of developer experience, for benefits I'm still not quite clear about. 

I agree with the principle of `separation by domain` but that does not imply that a stateless helper function should be part of a class. The principle is about organising code into logical units - Not enforcing object oriented programming paradigms when they are not useful. This function does not require any shared state so should be kept static. 

Furthermore, I noticed that we already have a collection of static helper functions for saved objects inside `SavedObjectsUtils` so I have added `generateId` and `isRandomId ` to that collection:
https://github.com/elastic/kibana/pull/84113/files#diff-7f0609a6fa9148304e94d7f5cb76ca7f1065adceee289ba5173b924a196f1d03

I hope this approach works for everyone. 

Let me know what you think.

Reply:
Yea, that will be some bigger refactor to expose this from the SO contract, and it would probably make sense for the core team to do it separately.

However in that case, I'd say that the best option may be to stick to using `uuid.v1` from the consuming code until we do implement this new API. @rudolf @restrry wdyt?

Reply:
> However in that case, I'd say that the best option may be to stick to using uuid.v1 from the consuming code until we do implement this new API. @rudolf @restrry wdyt?

:+1:

Reply:
> However in that case, I'd say that the best option may be to stick to using `uuid.v1` from the consuming code until we do implement this new API. @rudolf @restrry wdyt?

I believe the logic for ID generation is changing in the very near future (👀 @jportner), so I feel like it'd be safer to keep the logic consolidated in the static helper. Asking consumers to use `uuid.v1` is leaking an implementation detail of the SO service, which feels worse to me than exposing a static function from the service.

Reply:
> I believe the logic for ID generation is changing in the very near future (👀 @jportner)

Sharing Saved Objects will implement ID (re-)generation when existing objects are converted, but I have no plans to change "regular" ID generation.

Reply:
Let's keep it in `SavedObjectsUtils` then. it would be easier to track than greping for `uuid` anyway when we'll need to move that to the service contract.

Reply:
All done. Please give it another parse.